WEB-INF/classes/gov/noaa/pfel/erddap/jte/TableOptions.java

```diff
@@ -0,0 +1,225 @@
+    this.allowWrap = builder.allowWrap;
+    this.table = builder.table;
+    int nCols = table.nColumns();
+    this.fileAccessBaseUrl = new String[nCols];
+    this.fileAccessSuffix = new String[nCols];
+  }
+
+  public static class TableOptionsBuilder {
+    private String otherClasses = "";
+    private String bgColor = null;
+    private int border = 0;
+    private boolean writeUnits = false;
+    private int timeColumn = -1;
+    private boolean needEncodingAsHtml = false;
+    private boolean allowWrap = true;
+    private Table table;
+
+    public TableOptionsBuilder(Table table) {
+      this.table = table;
+    }
+
+    public TableOptionsBuilder otherClasses(String otherClasses) {
+      this.otherClasses = otherClasses;
+      return this;
+    }
+
+    public TableOptionsBuilder bgColor(String bgColor) {
+      this.bgColor = bgColor;
+      return this;
+    }
+
+    public TableOptionsBuilder border(int border) {
+      this.border = border;
+      return this;
+    }
+
+    public TableOptionsBuilder writeUnits(boolean writeUnits) {
+      this.writeUnits = writeUnits;
+      return this;
+    }
+
+    public TableOptionsBuilder timeColumn(int timeColumn) {
+      this.timeColumn = timeColumn;
+      return this;
+    }
+
+    public TableOptionsBuilder needEncodingAsHtml(boolean needEncodingAsHtml) {
+      this.needEncodingAsHtml = needEncodingAsHtml;
+      return this;
+    }
+
+    public TableOptionsBuilder allowWrap(boolean allowWrap) {
+      this.allowWrap = allowWrap;
+      return this;
+    }
+
+    public TableOptions build() {
+      return new TableOptions(this);
+    }
+  }
+
+  public String getNoData() {
+    return MustBe.THERE_IS_NO_DATA;
+  }
+
+  public String getWrap() {
+    return allowWrap ? "" : " nowrap";
+  }
+
+  public String getOtherClasses() {
+    return (otherClasses == null || otherClasses.length() == 0 ? "" : " " + otherClasses);
+  }
+
+  public String getBgColor() {
+    return bgColor == null ? "false" : bgColor;
+  }
+
+  public int getNColumns() {
+    return table.nColumns();
+  }
+
+  public int getNRows() {
+    return table.nRows();
+  }
+
+  public String insideForLoophead(int col) {
+    String s = table.getColumnName(col);
+    if (needEncodingAsHtml) {
+      s = XML.encodeAsHTML(s);
+    }
+    if (!somethingWritten && s.trim().length() > 0) {
+      somethingWritten = true;
+    } else if (col == getNColumns() - 1 && s.trim().length() == 0) {
+      s = "&nbsp;";
+    }
+
+    Attributes catts = table.columnAttributes(col);
+    fileAccessBaseUrl[col] = catts.getString("fileAccessBaseUrl");
+    fileAccessSuffix[col] = catts.getString("fileAccessSuffix");
+    if (!String2.isSomething(fileAccessBaseUrl[col])) fileAccessBaseUrl[col] = "";
+    if (!String2.isSomething(fileAccessSuffix[col])) fileAccessSuffix[col] = "";
+
+    return s;
+  }
+
+  public boolean getWriteUnits() {
+    return writeUnits;
+  }
+
+  public String insideForLoopUnits(int col) {
```

Comment:
```suggestion
  public String getColumnUnits(int col) {
```

Let's rename this method to `getColumnUnits`